src/Analysis/Codelyzer.Analysis.Build/WorkspaceBuilderHelper.cs

```diff
@@ -528,5 +541,116 @@ public void Dispose()
+            {
+                List<FileInfo> msBuildExePath = vsDirectory
+                    .GetDirectories("MSBuild", SearchOption.AllDirectories)
+                    .SelectMany(msBuildDir => msBuildDir.GetFiles("MSBuild.exe", SearchOption.AllDirectories))
+                    .OrderByDescending(msbuild => FileVersionInfo.GetVersionInfo(msbuild.FullName).FileVersion)
+                    .ThenBy(msbuild => editions.IndexOf(GetEditionType(msbuild.DirectoryName, editions)))
+                    .ThenBy(msbuild =>
+                    {
+                        var folderName = GetVersionFolder(msbuild.FullName);
+                        if (folderName.ToLower() == "current") return 1;
+                        else return -Convert.ToDouble(folderName);
+
+                    })
+                    .Where(msbuild =>
+                    {
+                        var targetsWithPath = GetTargetsWithPath(msbuild.DirectoryName, targets);
+                        if (targetsWithPath.TrueForAll(File.Exists)) return true;
+                        return false;
+                    })
+                    .ToList();
+                return msBuildExePath?.First().FullName;
+            };
+            return "";
+        }
+
+        public static string GetMsBuildPathFromVSDirectoryBelow15(DirectoryInfo vsDirectory, List<string> editions, string[] targets)
+        {
+            if (vsDirectory.Exists)
+            {
+                List<FileInfo> msBuildExePath = vsDirectory
+                    .GetFiles("MSBuild.exe", SearchOption.AllDirectories)
+                    .OrderByDescending(msbuild => FileVersionInfo.GetVersionInfo(msbuild.FullName).FileVersion)
+                    .ThenBy(msbuild =>
+                    {
+                        var folderName = GetVersionFolder(msbuild.FullName);
+                        if (folderName.ToLower() == "current") return 1;
+                        else return -Convert.ToDouble(folderName);
+
+                    })
+                    .Where(msbuild =>
+                    {
+                        var targetsWithPath = GetTargetsWithPath(msbuild.DirectoryName, targets);
+                        if (targetsWithPath.TrueForAll(File.Exists)) return true;
+                        return false;
+                    })
+                    .ToList();
+                return msBuildExePath?.First().FullName;
+            };
+            return "";
+        }
+
+        public static string GetEditionType(string vsPath, List<string> editions)
+        {
+            string[] elements = vsPath.ToLower().Split(@"\");
+            foreach (var edition in editions)
+            {
+                if (elements.Contains(edition.ToLower()))
+                {
+                    return edition;
+                }
+            }
+            return "";
+        }
+
+        public static string GetVersionFolder(string vsPath)
+        {
+            List<string> elements = vsPath.Split(@"\").ToList();
```

Comment:
You can use Path.DirectorySeparatorChar instead to make sure it's cross-platform